src/utilities/createRouterRoute.ts

```diff
@@ -1,6 +1,32 @@
 import { readonly } from 'vue'
-import { ResolvedRoute, RouterRoute } from '@/types'
+import { ResolvedRoute } from '@/types/resolved'
+import { RouteUpdate, RouteUpdateOptions, RouterUpdate } from '@/types/routerUpdate'
 
-export function createRouterRoute(route: ResolvedRoute): RouterRoute {
-  return readonly(route)
+export type RouterRoute<TRoute extends ResolvedRoute = ResolvedRoute> = Omit<ResolvedRoute, 'params'> & Readonly<{
+  params: { -readonly [P in keyof TRoute['params']]: TRoute['params'][P] },
```

Comment:
Nit: we should abstract this into a type utility called `Writable` or something similar.